### DISCUSSION TITLE
ist mein Modbus Konverter nutzbar?

### DISCUSSION THREAD

**Discussion by Dachmelker** [camueller/SmartApplianceEnabler#443] [Installation und Betrieb]
Hallo zusammen,

gerne würde ich mittels SAE den Verbrauch meiner Wärmepumpe im Sunny Portal visualisieren.

leider habe ich Schwierigkeiten mit dem Modbus, und es wird im Portal nur 0 angezeigt.

ich nutze diesen Konverter, wäre der generell nutzbar mit SAE ?
ein Test mit dem Simply Modbus Client war erfolgreich!

![Modbus Register](https://user-images.githubusercontent.com/122626002/212547256-9973d88a-8655-4f9a-9e4b-bc275f79051d.jpg)

![20220624_081512](https://user-images.githubusercontent.com/122626002/212546778-886cc6bb-61e9-47f9-89b9-c2aca72921ce.jpg)

**Comment by camueller** [comment]
Du hast in "Simply Modbus" als `Mode`  eingestellt `RTU over TCP`. Das bedeutet, dass Du einen RTU-Client hast, d.h. Du musst erstmal [`mbusd` installieren bzw. aktivieren](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Modbus_DE.md#modbus-protokoll).

Danach solltest Du nochmal mit "Simply Modbus"  testen, wobei als Mode `TCP` eingestellt ist. Wenn das erfolgreich ist, kannst Du mit der [Modbus-Konfiguration im SAE](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Settings_DE.md#modbus) weitermachen.

**Comment by Dachmelker** [comment]
das sieht schon vielversprechend aus, oder 

![sieht gut aus](https://user-images.githubusercontent.com/122626002/212558655-9e735dbe-b4ee-4188-ab15-068dde6d233b.jpg)

**Ist denn hier alles ok?**




![für github](https://user-images.githubusercontent.com/122626002/212558816-0b13adeb-5cb9-4101-ba49-e39309c4df57.jpg)

**Comment by camueller** [comment]
Passt denn der Zahlenwert  66210 zu Deinen Erwartungen? Eigentlich sollte ein Zählerstandswert mehr als 3 Nachkommastellen (wird normalerweise in Modbus als Float abgebildet) haben, sonst werden kleine Leisungen falsch berechnet. In diesem Fall kann es besser sein, die Leistung abzufragen anstatt des Zählerstandes.

Du solltest im SAE-Log checken, ob die Byte-Werte identisch sind mit denen im Simply Modbus.
Wenn das passt, kannst Du mit Anzahl der Datenwörter, Byte-Reihenfolge und Umrechnungsfaktor spielen, bis der umgewandelte Wert im Log in kWh ist.

**Comment by Dachmelker** [comment]
erstmal 1000 Dank, für deine Mühe 


Ja der Wert passt! 

662,10 kWh

Sorry für die doofe Frage, SAE Log finde ich wo?

![662](https://user-images.githubusercontent.com/122626002/212559619-d65bcef9-c0ca-4035-b158-d5d8d34b1948.jpg)

**Comment by camueller** [comment]
https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/Logging_DE.md

**Comment by Dachmelker** [comment]
Ich weiß nicht mehr weiter!

Ich bekomme einfach keine Werte im Sunny Portal angezeigt.

anbei mal das "semp"

kann das grün markierte evt das Problem sein?



![evt Prob](https://user-images.githubusercontent.com/122626002/212729596-9d462501-e454-40be-bc70-f1e5d6d1101b.jpg)

**Comment by camueller** [comment]
Dein Geräte soll ja vom SHM nicht gesteuert werden, also ist `EMSignalsAccepted=false` völlig in Ordnung.

Bevor Du im Sunny Portal oder unter /semp nach den werten schaust, solltest Du im SAE-Log prüfen, was der SAE aus dem Register ausliest und wie er es konvertiert (hatte ich gestern schon geschrieben).

**Comment by Dachmelker** [comment]
Mensch, ich bin wirklich dankbar für deinen Support 👍 

also Werte kommen im Log an, zwar ohne Kommastellen aber ich vermute das lässt sich lösen. ( transformed value 67026) 
670,26 kWh ist der echte Zählerwert

es kommt aber eine Warnung: **Meter not set**

ich habe hier im Forum gelesen, das das mit einem nicht gesetzten Schalter zu tun haben könnte?

ich denke wir sind auf den richtigen Weg.

**Comment by camueller** [comment]
> also Werte kommen im Log an, zwar ohne Kommastellen aber ich vermute das lässt sich lösen. ( transformed value 67026)
670,26 kWh ist der echte Zählerwert

Dann musst Du beim Zähler im SAE nur einen Umrechnugnsfaktor von 0.01 angeben

> es kommt aber eine Warnung: Meter not set. ich habe hier im Forum gelesen, das das mit einem nicht gesetzten Schalter zu tun haben könnte?

Es muss kein Schalter gesetzt werden. Ist auch nur eine Warnung und kein Fehler.

**Reply by Dachmelker** [reply]
ok 👍 

aber warum wird im SunnyPortal dann das Gerät mit **0W** angezeigt.?

Und wenn ich versuche im SAE unter Zähler einen Faktor anzugeben, dann wird kein Punkt, und auch kein Komma akzeptiert.

heute Abend bin ich wieder daheim,

**Comment by Dachmelker** [comment]
Ich habe Blödsinn erzählt.

ein Punkt beim Umrechnungsfaktor ist doch möglich.

leider "laggt" das SunnyPortal  sehr stark, zur Zeit 1,25 Std zurück.

Ich hoffe das dann jetzt auch Werte im Portal ankommen.

Im Augenblick ist da noch nichts

**Comment by camueller** [comment]
Du musst nicht auf das Portal warten bzw. Du solltest an der [SEMP-Schnittstelle des SAE](https://github.com/camueller/SmartApplianceEnabler/blob/master/doc/SEMP_DE.md#semp-url) prüfen, ob diese die richtige Leistung übermittelt an den SHM. Alternativ kannst Du auch im Log nachsehen.

**Comment by Dachmelker** [comment]
Klasse, wie schnell du bist 

aber leider Nein :(

das ist bestimmt nur eine Kleinigkeit, die ich nicht sehe 

![semp 17 01](https://user-images.githubusercontent.com/122626002/212947496-81c422a4-986c-433e-bf60-19d64c02ba34.jpg)

**Reply by Dachmelker** [reply]
Im Log wurde der Wert auch nicht umgerechnet.

Also der Wert ist immer noch ohne Kommastelle

**Comment by camueller** [comment]
Dann poste doch mal diese Zeilen aus dem Log:
```
2023-01-13 11:01:33,534 TRACE [Timer-0] d.a.s.m.e.ReadInputRegisterExecutor [ReadInputRegisterExecutor.java:42] F-00000001-000000000019-00: Reading input register=342 requestWords=2
2023-01-13 11:01:33,585 DEBUG [Timer-0] d.a.s.m.e.ReadInputRegisterExecutor [ReadInputRegisterExecutor.java:56] F-00000001-000000000019-00: Input register=342 value=[17799, 28486]
2023-01-13 11:01:33,586 DEBUG [Timer-0] d.a.s.m.t.FloatValueTransformer [FloatValueTransformer.java:44] F-00000001-000000000019-00: transformed value=4333.9091796875
```

**Comment by Dachmelker** [comment]
screenshot mal anders 

![SAE Screenshot klein](https://user-images.githubusercontent.com/122626002/212961662-2c3e5b3e-8eee-40de-98ed-ce47866226c8.jpg)

**Comment by camueller** [comment]
Ich habe gerade im Code nachgesehen: Für Integer-Werte wird aktuell der Umrechnungsfaktor nicht angewendet. Mit der aktuellen SAE-Version kommst Du also nicht weiter.

**Comment by Dachmelker** [comment]
ok, schade.

gibt es Hoffnung auf ein baldiges Update

**Comment by camueller** [comment]
Mal schauen. Je mehr Zeit für Support draufgeht, umso weniger bliebt für Code-Anpassungen.

**Reply by Dachmelker** [reply]
ich drück die Daumen.

auch wenn es bei mir jetzt nicht funktioniert hat, aber deine Arbeit und Einsatz für das Projekt sind absolut bemerkenswert.

Lob, Respekt & Anerkennung

**Comment by camueller** [comment]
Habe mir Deinen Screenshot nochmal genauer angesehen: Du hast als Wert-Typ "Integer" eingestellt. Stell mal "Float als Integer" ein. Das klingt zwar falsch, sollte aber funktionieren. Hat es zumindest gerade bei einem Test hier.

**Comment by Dachmelker** [comment]
![grün](https://user-images.githubusercontent.com/122626002/212981989-1be01d97-5744-4db6-bdf5-88bcf6ddac0c.jpg)
 
💯

**Comment by camueller** [comment]
Wenn 600W zu Deiner Erwartung passen, ist ja alls gut :-)

**Comment by Dachmelker** [comment]
Ich bin happy 
um die 600 Watt kann passen. 

ich prüfe das aber noch.


bis die Werte im Portal sichtbar sind, dauert es warscheinlich, oder

**Comment by Dachmelker** [comment]
LÄUFT 💯 Perfekt

ich habe aber auf aktuelle Leistung umgestellt, weil es genauer ist.


![Screenshot 2023-01-17 at 20-18-32 Verbraucherbilanz und -steuerung - Dachmelkers EV Anlage - Sunny Portal](https://user-images.githubusercontent.com/122626002/212992351-283ca7ad-67c7-4e32-ac50-388f7c90e768.png)

**Comment by Dachmelker** [comment]
Hallo Axel, Hallo zusammen

das SAE läuft einwandfrei, soweit alles gut


Jetzt finde ich im Sunny Portal nicht mehr die Anzeige, wo der Verbrauch der einzelnen Geräte angezeigt wird.

Also mit Anteil des Eigenstroms in Prozent.

**Reply by mivola** [reply]
> Jetzt finde ich im Sunny Portal nicht mehr die Anzeige, wo der Verbrauch der einzelnen Geräte angezeigt wird.
> 
> Also mit Anteil des Eigenstroms in Prozent.

Schau mal in die Tabelle unter dem Monats/Jahresdiagramm: https://www.sunnyportal.com/Homan/ConsumerBalance#/month/1675728000000

**Comment by Dachmelker** [comment]
Hi,  

danke

dann war ich da ja schon richtig.

Leider sieht das bei mir so aus 



![Problem Sunny Portal](https://user-images.githubusercontent.com/122626002/217290480-26b1cc55-5e38-4147-9a65-f42f89aa9587.jpg)

**Comment by Dachmelker** [comment]
Problem gefunden.

ich hatte ein Zeitfenster zum Batterieladen aktiv.

in diesem Fall blendet das Sunny Portal die Angaben für den Eigenverbrauch aus.
